prov/util/src/util_mr_cache.c

```diff
@@ -364,6 +358,19 @@ static struct ofi_mr_entry *ofi_mr_rbt_find(struct ofi_mr_storage *storage,
 	return node->data;
 }
 
+static struct ofi_mr_entry *ofi_mr_rbt_overlap(struct ofi_mr_storage *storage,
+					    const struct iovec *key)
+{
+	struct ofi_rbnode *node;
+
+	node = ofi_rbmap_search(storage->storage, (void *) key,
```

Comment:
This still does not resolve the issue of adjacent buffers.

Ex. Find overlap will match buffers A and B if:
A: iov_base = 0x0, iov_len = 1
B: iov_base = 0x1, iov_len = 1
(Simplified, but multiply by page size)
If B experiences a page fault, it should not matter to A. We need to exclude the adjacent buffer match here.

Reply:
Yes, this will currently match and uncache adjacent buffers incorrectly.  That shouldn't affect correctness, however, only performance.  We just need to introduce a new match function.

Reply:
Yes, this should work as is. As soon as the new match function is introduced, I will approve this PR.